### PR TITLE
Add TTC streetcar network +  Tramway de Québec

### PR DESCRIPTION
**Pull request by acquahki**
- Addition of the TTC streetcar network:

> **Routes**
> - 501
> - 503
> - 504
> - 505
> - 506
> - 509
> - 510
> - 511
> - 512
> 
> **Loops**
> - Bingham Loop
> - Charlotte Loop
> - College Loop
> - Coxwell Loop
> - Distillery Loop
> - Dufferin Gate Loop
> - Earlscourt Loop
> - Exhibition Loop
> - Fleet Loop
> - Gunns Loop
> - High Park Loop
> - Humber Loop
> - Kipling Loop
> - Long Branch Loop
> - McCaul Loop
> - Neville Park Loop
> - Oakwood Loop
> - Queens Quay Loop
> - Sunnyside Loop
> - Wolseley Loop
> - Woodbine Loop
> 
> **MSFs**
> - Hillcrest Complex
> - Roncesvalles Carhouse
> - Russell Carhouse
> - Leslie Barns
> 
> **Connector Tracks**
> - Adelaide
> - Bathurst Connector
> - Bay
> - Church
> - Coxwell
> - Dufferin
> - McCaul
> - Ossington
> - Parliament
> - Richmond
> - Shaw
> - Victoria
> - York

- ION track fix north of Allen station
- Added the proposed alignment for the Tramway de Québec / Quebec City Tramway.

### CODE REVIEW

**Review by dellisd — CHANGES_REQUESTED** [review]
First of all, this is some amazing work and is a huuuuuuuge contribution to the map!!

I honestly didn't realize just how many connectors and loops there are (woops) so I think we can combine all of the connector tracks into a single file, and all of the loops into a single file. Also, can you move all of the streetcar-related files into a `streetcar` subdirectory under `data/toronto`? None of the IDs need to change, but this will help keep things organized.

Lastly, the streetcar network adds a lot of information density to the map. Using the exact same styling as the other rail lines ends up being really cluttered (and this is still without even adding GO to the map) so I'm experimenting with some alternate styles for the streetcar lines. What I think will happen is that I'll open a separate PR with the changes to add the new style layers, and then this PR can be merged on top of it to make use of the new styling.

Here's what I've been trying so far:
image

If you have any ideas on how the styling can be improved or any other thoughts, please let me know!

Again, this is fantastic!!! 🚀

File: data/toronto/route510.json [hunk 1]
```diff
@@ -0,0 +1,80 @@
+{
```
[the remark this answers is not in the document]
Comment (acquahki) [reply data/toronto/route510.json:1]: Added tracks in `toronto/streetcarConnectors` at the following streets:
- Spadina / College
- Spadina / Dundas
- Spadina / Queen
- Spadina / King
- Spadina / Queen's Quay
- Bathurst / College
- Bathurst / Dundas
- Bathurst / Queen
- Bathurst / King
- Bathurst / Lake Shore
- Queensway/King/Queen
- Roncesvalles / Howard Park
- King / Sumach
- Broadview / Queen
- Broadview / Gerrard
- Broadview / Dundas
- Coxwell / Gerrard

**Comment by acquahki** [comment]
> I honestly didn't realize just how many connectors and loops there are (woops) so I think we can combine all of the connector tracks into a single file, and all of the loops into a single file. Also, can you move all of the streetcar-related files into a `streetcar` subdirectory under `data/toronto`?

`streetcarLoops.json` and `streetcarConnectors.json` have been created in the `data/toronto/streetcar` subdirectory.


> Here's what I've been trying so far (thinner lines and different stop label styling): 

The new layer styling look great! Greatly helps differentiate the streetcar infrastructure in the city.

**Comment by dellisd** [comment]
For any loop that is always used in service (e.g. Gunns loop, or High Park loop) I think we can omit the extra line geometry, but keep the labels for the loop.
image

**Review by dellisd — APPROVED** [review]

**Pull request merged**